connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/ProtobufOptions.scala

```diff
@@ -39,15 +39,72 @@ private[sql] class ProtobufOptions(
+   *  1:  `struct<name: string>`
+   *  2:  `struct<name string, friend: struct<name: string>>`
+   *  3:  `struct<name string, friend: struct<name string, friend: struct<name: string>>>`
+   * and so on.
+   */
   val recursiveFieldMaxDepth: Int = parameters.getOrElse("recursive.fields.max.depth", "-1").toInt
 
-  // Whether to render fields with zero values when deserializing Protobufs to a Spark struct.
+  /**
+   * This option enables converting Protobuf 'Any' fields to JSON. At runtime, such 'Any' fields
+   * can contain arbitrary Protobuf messages as binary data.
+   *
+   * By default when this option is not enabled, such field behaves like normal Protobuf message
+   * with two fields (`STRUCT<type_url: STRING, value: BINARY>`). The binary `value` field is not
+   * interpreted. The binary data might not be convenient in practice to work with.
+   *
+   * One option is to deserialize it into actual Protobuf message and convert it to Spark STRUCT.
+   * But this is not feasible since the schema for `from_protobuf()` is needed at query compile
+   * time and can not change at runtime. As a result, this option is not feasible.
+   *
+   * Another option is parse the binary and deserialize the Protobuf message into JSON string.
+   * This this lot more readable than the binary data. This configuration option enables
+   * converting Any fields to JSON. The example blow clarifies further.
+   *
+   *  Consider two Protobuf types defined as follows:
+   *    message ProtoWithAny {
+   *       string event_name = 1;
+   *       google.protobuf.Any details = 2;
+   *    }
+   *
+   *    message Person {
+   *      string name = 1;
+   *      int32 id = 2;
+   *   }
+   *
+   * With this option enabled, schema for `from_protobuf("col", messageName = "ProtoWithAny")`
+   * would be : `STRUCT<event_name: STRING, details: STRING>`.
+   * At run time, if `details` field contains `Person` Protobuf message, the returned value looks
+   * like this:
+   *   ('click', '{"@type":"type.googleapis.com/...ProtoWithAny","name":"Mario","id":100}')
+   *
+   * Requirements:
+   *  - The definitions for all the possible Protobuf types that are used in Any fields should be
+   *    available in the Protobuf descriptor file passed to `from_protobuf()`. If any Protobuf
+   *    is not found, it will result in error for that record.
+   *  - This feature is supported with Java classes as well. But only the Protobuf types defined
+   *    in the same `proto` file as the primary Java class might be visible.
+   *    E.g. if `ProtoWithAny` and `Person` in above example are in different proto files,
+   *    definition for `Person` may not be found.
+   *
+   * This feature should be enabled carefully. JSON conversion and processing are inefficient.
+   * In addition schema safety is also reduced making downstream processing error prone.
+   */
+  val convertAnyFieldsToJson: Boolean =
+    parameters.getOrElse("convert.any.fields.to.json", "false").toBoolean
```

Comment:
@rangadi let's use the option as `convertAnyFieldToJson`. Also, let's put it in a constant variable instead of hard-coding it across the code.
Thus naming style is consistent with other data sources like csv/json:
https://spark.apache.org/docs/latest/sql-data-sources-csv.html
https://spark.apache.org/docs/latest/sql-data-sources-json.html

Reply:
This is not a source, btw. We already started using spark config naming style. I would like to keep this convention if I can.
I can move it to a constant in a follow up PR in order to cut down CI/CD time.

Reply:
There is a large scaladoc comment